### DISCUSSION TITLE
Question on auto routering via STOW

### DISCUSSION THREAD

**Discussion by luxunxiansheng** [dcm4che/dcm4chee-arc-light#3029] [Q&A] [answered]
Hi, 
     Here is our case
     
     We need a compact installation on the IOT-like device, say , a  
     not very powerful mini computer sitting at the rural hospital ,  
     to auto router the received instances to cloud via STOW.  

     1) We know Dcm4chee-arc-light supports the auto routering via DICOM  
         and we are trying to config the setup to forward the instances from scu 
         to cloud via internet. If anyone can tell us in advance the configuration 
         which may save us much time.  

    2) Suppose we are targeting a mini computer,  is there any most comact 
         configuration to make the router work?

Many thanks 

L.B.

**Comment by gunterze** [comment, marked as answer]
There is already an open issue for implementing export by STOW-RS (#2836). Hope we can implement it in next 2 months.
You may workaround using [DUL STORE/STOW-RS Gateway Karnark](https://github.com/OsiriX-Foundation/karnak).

**Reply by luxunxiansheng** [reply]
Wow, that's awesome!

**Reply by luxunxiansheng** [reply]
BTW: From our case , we need the router  to run in a compact mini-box, so it is highly 
desired for dcm4chee to give us a guideline  how to config such a working env.  where 
the resource is limited.    

Thank you so much.

**Reply by gunterze** [reply]
If you are just interested in forwarding DICOM objects received from any DUL C-STORE SCU - particularly modalities -, without the need of local (intermediate) storage and query/retrieve from/to other DICOM peer applications - particularly image displays -, using plain Karnak, or connecting dcm4che's CL utilities [storescp](https://github.com/dcm4che/dcm4che/blob/master/dcm4che-tool/dcm4che-tool-storescp/README.md) and [stowrs](https://github.com/dcm4che/dcm4che/blob/master/dcm4che-tool/dcm4che-tool-stowrs/README.md) by scripts or your own java application, may be more appropriate than a full fledged DICOM archive as dcm4chee-arc.

**Reply by luxunxiansheng** [reply]
In our case,  we have to transfer the pacs data (which may be a heavy work especially during the busy time)  within intranet to  cloud fast and realiably. we love dcm4chee-arc for its stability and industral strength. We believe it has better machanism to handle ,say, error-handling, concurrency or something like this than us to do.

Anyway，we will test karnak but still perfer a more light   dcm4chee-router in the future

**Reply by luxunxiansheng** [reply]
Hi, Gunterze, 
     We have played  the karnak for a while. It seems not doing well at this moment as a new  project.  Do you have other recommendations  for a light, reliable  storescp-stow  gateway ?   At the same time, we also highly suggest dcm4chee-router can take our case into account . It is actually very typical  IoT solution.

Best Regards
L.B.

**Comment by luxunxiansheng** [comment]
In case someone is also seeking the same solution as us, there is a open source project that seems good .

https://github.com/GoogleCloudPlatform/healthcare-dicom-dicomweb-adapter
